libs/ai-endpoints/langchain_nvidia_ai_endpoints/chat_models.py

```diff
@@ -510,16 +518,214 @@ def bind_functions(
     ) -> Runnable[LanguageModelInput, BaseMessage]:
         raise NotImplementedError("Not implemented, use `bind_tools` instead.")
 
-    def with_structured_output(
+    # we have an Enum extension to BaseChatModel.with_structured_output and
+    # as a result need to type ignore for the schema parameter and return type.
+    def with_structured_output(  # type: ignore
         self,
-        schema: _DictOrPydanticClass,
+        schema: _DictOrPydanticOrEnumClass,
         *,
-        method: Literal["function_calling", "json_mode"] = "function_calling",
```

Comment:
Do I understand this correctly, and we decided remove the ability to specify method?
https://python.langchain.com/v0.2/docs/how_to/structured_output/#advanced-specifying-the-method-for-structuring-outputs

Reply:
I could be wrong, but specifying JSON mode should force it output JSON regardless. Also this would be consistent with broader lc usage.

Reply:
the `method` param effectively requires the user to pick the implementation that is used to achieve their desired outcome of json or pydantic responses from `invoke`/`stream`. we can do better than requiring the user to pick.

also, https://python.langchain.com/v0.2/docs/how_to/structured_output/#advanced-specifying-the-method-for-structuring-outputs
![image](https://github.com/user-attachments/assets/77c345d4-73d9-4777-bfed-ebc44b28e062)

"JSON mode" asserts it will not pass the schema to the model, but we will pass the schema to the model.

Reply:
ok sgtm!